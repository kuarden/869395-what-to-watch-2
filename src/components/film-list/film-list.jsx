import React, {PureComponent} from "react";
import {arrayOf, func, shape, string, number} from 'prop-types';
import FilmCard from '../film-card/film-card.jsx';
import withActiveItem from "../../hocs/with-active-item";

export class FilmList extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {films, onHeaderClick, onChangeActiveItem = () => ({})} = this.props;
    return <div className="catalog__movies-list">{films.map((film, i) =>
      <FilmCard key = {`movie-${film.id}`}
        film = {film}
        onHeaderClick = {onHeaderClick}
        onMouseEnter= {() => onChangeActiveItem(i)}
        onMouseLeave = {() => onChangeActiveItem(-1)}>
      </FilmCard>
    )}</div>;
  }
}

FilmList.propTypes = {
  films: arrayOf(shape({
    id: number,
    name: string,
    genre: string,
    previewImage: string,
    videoLink: string,
  })).isRequired,
  onHeaderClick: func.isRequired,
  activeItem: number,
  onChangeActiveItem: func,
};

export default withActiveItem(FilmList);
