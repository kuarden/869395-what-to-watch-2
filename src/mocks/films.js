export const films = [
  {
    id: 1,
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Fantasy`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 2,
    name: `Bohemian Rhapsody`,
    image: `img/bohemian-rhapsody.jpg`,
    genre: `Drama`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 3,
    name: `Macbeth`,
    image: `img/macbeth.jpg`,
    genre: `Fantasy`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 4,
    name: `Aviator`,
    image: `img/aviator.jpg`,
    genre: `Adventures`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 5,
    name: `We need to talk about Kevin`,
    image: `img/we-need-to-talk-about-kevin.jpg`,
    genre: `Adventures`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 6,
    name: `What We Do in the Shadows`,
    image: `img/what-we-do-in-the-shadows.jpg`,
    genre: `Fantasy`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 7,
    name: `Revenant`,
    image: `img/revenant.jpg`,
    genre: `Drama`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 8,
    name: `Johnny English`,
    image: `img/johnny-english.jpg`,
    genre: `Comedy`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 9,
    name: `Shutter Island`,
    image: `img/shutter-island.jpg`,
    genre: `Thriller`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 10,
    name: `Pulp Fiction`,
    image: `img/pulp-fiction.jpg`,
    genre: `Action`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 11,
    name: `No Country for Old Men`,
    image: `img/no-country-for-old-men.jpg`,
    genre: `Action`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 12,
    name: `Snatch`,
    image: `img/snatch.jpg`,
    genre: `Action`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 13,
    name: `Moonrise Kingdom`,
    image: `img/moonrise-kingdom.jpg`,
    genre: `Fantasy`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 14,
    name: `Seven Years in Tibet`,
    image: `img/seven-years-in-tibet.jpg`,
    genre: `Adventures`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 15,
    name: `Midnigth Special`,
    image: `img/midnight-special.jpg`,
    genre: `Fantasy`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 16,
    name: `War of the Worlds`,
    image: `img/war-of-the-worlds.jpg`,
    genre: `Fantasy`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 17,
    name: `Dardjeeling Limited`,
    image: `img/dardjeeling-limited.jpg`,
    genre: `Adventures`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 18,
    name: `Orlando`,
    image: `img/orlando.jpg`,
    genre: `Adventures`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 19,
    name: `Mindhunter`,
    image: `img/mindhunter.jpg`,
    genre: `Action`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 20,
    name: `Midnight Special`,
    image: `img/midnight-special.jpg`,
    genre: `Fantasy`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
];
