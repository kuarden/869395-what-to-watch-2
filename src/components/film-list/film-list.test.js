import React from "react";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import {FilmList} from "./film-list.jsx";
import {films} from "../../mocks/films.js";

Enzyme.configure({adapter: new Adapter()});

it(`FilmList renders correctly`, () => {
  const tree = shallow(<FilmList
    films={films}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
