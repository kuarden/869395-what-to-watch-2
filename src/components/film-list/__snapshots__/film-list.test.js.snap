// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <FilmCard
    film={
      Object {
        "genre": "Fantasy",
        "id": 1,
        "image": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "name": "Fantastic Beasts: The Crimes of Grindelwald",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    handleMouseOver={[Function]}
    key="1 0"
  />
  <FilmCard
    film={
      Object {
        "genre": "Drama",
        "id": 2,
        "image": "img/bohemian-rhapsody.jpg",
        "name": "Bohemian Rhapsody",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    handleMouseOver={[Function]}
    key="2 1"
  />
  <FilmCard
    film={
      Object {
        "genre": "Fantasy",
        "id": 3,
        "image": "img/macbeth.jpg",
        "name": "Macbeth",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    handleMouseOver={[Function]}
    key="3 2"
  />
  <FilmCard
    film={
      Object {
        "genre": "Adventures",
        "id": 4,
        "image": "img/aviator.jpg",
        "name": "Aviator",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    handleMouseOver={[Function]}
    key="4 3"
  />
  <FilmCard
    film={
      Object {
        "genre": "Adventures",
        "id": 5,
        "image": "img/we-need-to-talk-about-kevin.jpg",
        "name": "We need to talk about Kevin",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    handleMouseOver={[Function]}
    key="5 4"
  />
  <FilmCard
    film={
      Object {
        "genre": "Fantasy",
        "id": 6,
        "image": "img/what-we-do-in-the-shadows.jpg",
        "name": "What We Do in the Shadows",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    handleMouseOver={[Function]}
    key="6 5"
  />
  <FilmCard
    film={
      Object {
        "genre": "Drama",
        "id": 7,
        "image": "img/revenant.jpg",
        "name": "Revenant",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    handleMouseOver={[Function]}
    key="7 6"
  />
  <FilmCard
    film={
      Object {
        "genre": "Comedy",
        "id": 8,
        "image": "img/johnny-english.jpg",
        "name": "Johnny English",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    handleMouseOver={[Function]}
    key="8 7"
  />
  <FilmCard
    film={
      Object {
        "genre": "Thriller",
        "id": 9,
        "image": "img/shutter-island.jpg",
        "name": "Shutter Island",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    handleMouseOver={[Function]}
    key="9 8"
  />
  <FilmCard
    film={
      Object {
        "genre": "Action",
        "id": 10,
        "image": "img/pulp-fiction.jpg",
        "name": "Pulp Fiction",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    handleMouseOver={[Function]}
    key="10 9"
  />
  <FilmCard
    film={
      Object {
        "genre": "Action",
        "id": 11,
        "image": "img/no-country-for-old-men.jpg",
        "name": "No Country for Old Men",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    handleMouseOver={[Function]}
    key="11 10"
  />
  <FilmCard
    film={
      Object {
        "genre": "Action",
        "id": 12,
        "image": "img/snatch.jpg",
        "name": "Snatch",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    handleMouseOver={[Function]}
    key="12 11"
  />
  <FilmCard
    film={
      Object {
        "genre": "Fantasy",
        "id": 13,
        "image": "img/moonrise-kingdom.jpg",
        "name": "Moonrise Kingdom",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    handleMouseOver={[Function]}
    key="13 12"
  />
  <FilmCard
    film={
      Object {
        "genre": "Adventures",
        "id": 14,
        "image": "img/seven-years-in-tibet.jpg",
        "name": "Seven Years in Tibet",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    handleMouseOver={[Function]}
    key="14 13"
  />
  <FilmCard
    film={
      Object {
        "genre": "Fantasy",
        "id": 15,
        "image": "img/midnight-special.jpg",
        "name": "Midnigth Special",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    handleMouseOver={[Function]}
    key="15 14"
  />
  <FilmCard
    film={
      Object {
        "genre": "Fantasy",
        "id": 16,
        "image": "img/war-of-the-worlds.jpg",
        "name": "War of the Worlds",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    handleMouseOver={[Function]}
    key="16 15"
  />
  <FilmCard
    film={
      Object {
        "genre": "Adventures",
        "id": 17,
        "image": "img/dardjeeling-limited.jpg",
        "name": "Dardjeeling Limited",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    handleMouseOver={[Function]}
    key="17 16"
  />
  <FilmCard
    film={
      Object {
        "genre": "Adventures",
        "id": 18,
        "image": "img/orlando.jpg",
        "name": "Orlando",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    handleMouseOver={[Function]}
    key="18 17"
  />
  <FilmCard
    film={
      Object {
        "genre": "Action",
        "id": 19,
        "image": "img/mindhunter.jpg",
        "name": "Mindhunter",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    handleMouseOver={[Function]}
    key="19 18"
  />
  <FilmCard
    film={
      Object {
        "genre": "Fantasy",
        "id": 20,
        "image": "img/midnight-special.jpg",
        "name": "Midnight Special",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      }
    }
    handleMouseOver={[Function]}
    key="20 19"
  />
</div>
`;
